tbx/navigation/models.py

```diff
@@ -100,3 +36,17 @@ class NavigationSettings(BaseSiteSetting, ClusterableModel):
         FieldPanel("footer_links"),
         FieldPanel("footer_logos"),
     ]
+
+    def save(self, **kwargs):
+        super().save(**kwargs)
+
+        fragment_keys = ["primarynav"]
+        keys = [
+            # The fragment cache varies on:
+            # the current site pk, whether used in preview, or in the pattern library
+            # NOTE: `is_pattern_library` returns True if pattern is being rendered in the pattern library,
```

Comment:
Have you set `is_pattern_library: True` in the yaml? This is a convention we use rather than a value that always exists. We might need to set it on all page template yaml configs, and for any other templates like the header that include the navigation, in order for it to work consistently. I'm happy to look at this as part of the front-end.

Reply:
Thanks for the clarification and for offering to look at it as part of the FE PR!

From the [docs](https://torchbox.github.io/django-pattern-library/reference/api/#is_pattern_library), I got the impression that it always existed, but I may have misunderstood it :slightly_smiling_face: :

> `is_pattern_library` is available in the template context of each pattern, and is `True` if the pattern is being rendered in the pattern library.

Reply:
Oh I had no idea about that! We must have taken that convention and made it part of the core functionality. Good to know.